testsuite/features/step_definitions/common_steps.rb

```diff
@@ -643,25 +643,31 @@
   $server.run("salt-key -y --accept=pxeboot.example.org")
 end
 
-When(/^I bootstrap (traditional|minion) client "([^"]*)" using bootstrap script with activation key "([^"]*)" from the proxy$/) do |client_type, host, key|
+When(/^I bootstrap (traditional|minion) client "([^"]*)" using bootstrap script with activation key "([^"]*)" from the server|proxy$/) do |client_type, host, key, target_type|
   # Preparation of bootstrap script for different types of clients
   client = client_type == 'traditional' ? '--traditional' : ''
+  # Use server if proxy is not defined as proxy is not mandatory
+  if target_type.include? 'server' || $proxy.nil?
+    target = $server
+  else
+    target = $proxy
+  end
   cmd = "mgr-bootstrap #{client} &&
   sed -i s\'/^exit 1//\' /srv/www/htdocs/pub/bootstrap/bootstrap.sh &&
   sed -i '/^ACTIVATION_KEYS=/c\\ACTIVATION_KEYS=#{key}' /srv/www/htdocs/pub/bootstrap/bootstrap.sh &&
   chmod 644 /srv/www/htdocs/pub/RHN-ORG-TRUSTED-SSL-CERT &&
   sed -i '/^ORG_GPG_KEY=/c\\ORG_GPG_KEY=RHN-ORG-TRUSTED-SSL-CERT' /srv/www/htdocs/pub/bootstrap/bootstrap.sh &&
   cat /srv/www/htdocs/pub/bootstrap/bootstrap.sh"
-  output, code = $proxy.run(cmd)
+  output, code = target.run(cmd)
   raise "Key: #{key} not included" unless output.include? key
   # Run bootstrap script and check for result
   boostrap_script = 'bootstrap-general.exp'
   source = File.dirname(__FILE__) + '/../upload_files/' + boostrap_script
   dest = '/tmp/' + boostrap_script
-  return_code = file_inject($proxy, source, dest)
+  return_code = file_inject(target, source, dest)
   raise 'File injection failed' unless return_code.zero?
   system_name = get_system_name(host)
-  output, code = $proxy.run("expect -f /tmp/#{boostrap_script} #{system_name}")
+  output, code = target.run("expect -f /tmp/#{boostrap_script} #{system_name}")
```

Comment:
Same here (unused `code`).

Reply:
True, I will change it. Just at the beginning I had plan to check code value to be equal to 6. I abandoned the idea, but this remained. Good catch :-)